cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/inmemory/InMemoryMetricsTable.java

```diff
@@ -39,6 +41,16 @@
 
   private final String tableName;
 
+  /**
+   * To be used in tests that need namespaces
+   */
+  public InMemoryMetricsTable(DatasetContext datasetContext, String name, CConfiguration cConf) {
+    this(PrefixedNamespaces.namespace(cConf, datasetContext.getNamespaceId(), name));
+  }
+
+  /**
+   * To be used in tests that do not need namespaces
+   */
```

Comment:
I think it would be better to remove this completely. Anybody who is creating a table should be namespace aware.